src/extensions/life_time_extension.rs

```diff
@@ -49,30 +50,48 @@ impl LifetimeExtension {
             not_after,
         }
     }
-    pub fn new_from_bytes(bytes: &[u8]) -> Self {
+
+    fn is_expired(&self) -> bool {
+        let now = SystemTime::now()
+            .duration_since(UNIX_EPOCH)
+            .unwrap()
+            .as_secs();
+        self.not_before < now && self.not_after > now
```

Comment:
I think this actually `is_valid()`...

Reply:
Possible... I'll double-check the logic here. This will also get a test